In [1]:
import numpy as np
print("NumPy is working")

NumPy is working


## Smart Electricity Consumption Analyzer (Using NumPy Arrays)

Used
- Python
- NumPy (arrays only)

Data Assumption
- 1 month = 30 days
- Daily electricity units consumed

In [3]:
units = np.array([
    5.2, 6.1, 5.8, 7.0, 8.5,
    9.2, 6.4, 5.9, 7.8, 8.1,
    6.5, 7.3, 9.0, 10.2, 8.9,
    7.6, 6.8, 5.5, 6.0, 7.1,
    8.4, 9.6, 10.5, 9.8, 8.2,
    7.0, 6.3, 5.9, 6.7, 7.4
])                               # 1D Array

print("Daily Units:", units)

Daily Units: [ 5.2  6.1  5.8  7.   8.5  9.2  6.4  5.9  7.8  8.1  6.5  7.3  9.  10.2
  8.9  7.6  6.8  5.5  6.   7.1  8.4  9.6 10.5  9.8  8.2  7.   6.3  5.9
  6.7  7.4]


#### Total & Average Consumption

In [6]:
total_units = np.sum(units) # sum of all units 
avg_units = np.mean(units)   # average units
med_units = np.median(units)   # median units
min_units = np.min(units)   # minimum units
max_units = np.max(units)   # maximum units

print("Total Units in Month:", total_units)
print("Average of Daily Units:", avg_units)
print("Median of Daily Units:", med_units)
print("Minimum of Daily Units:", min_units)
print("Maximum of Daily Units:", max_units)



Total Units in Month: 224.7
Average of Daily Units: 7.489999999999999
Median of Daily Units: 7.199999999999999
Minimum of Daily Units: 5.2
Maximum of Daily Units: 10.5


- Minimum unit is 5.2 , and maximum is 10.5.
- the average unit is around 7.48.
- the month units is 224.7.


#### High Usage Days (>9 units)

In [12]:
high_usage_days = np.where(units > 9)

print("High Usage Days:", high_usage_days[0] + 1)


High Usage Days: [ 6 14 22 23 24]


#### Wastage Detection
- (If usage > 20% above average)

In [8]:
wastage_days = np.where(units > avg_units * 1.2)

print("Possible Wastage Days:", wastage_days[0] + 1)


Possible Wastage Days: [ 6 13 14 22 23 24]


#### Electricity Bill Estimation

- Assume:
1 unit = ₹6

In [9]:
bill_amount = total_units * 6
print("Estimated Electricity Bill: ₹", bill_amount)

Estimated Electricity Bill: ₹ 1348.1999999999998


#### Best & Worst Day

In [10]:
best_day = np.argmin(units) + 1
worst_day = np.argmax(units) + 1

print("Lowest Consumption Day:", best_day)
print("Highest Consumption Day:", worst_day)


Lowest Consumption Day: 1
Highest Consumption Day: 23


### Final Project Summary
- Analyzed 30 days of electricity consumption using NumPy 1D arrays
- Performed descriptive statistics to understand usage behavior
- Identified high-usage and potential wastage days
- Estimated monthly electricity cost based on unit pricing

### Key Metrics

- Total Units Consumed: 224.7 units
- Average Daily Consumption: ~7.48 units
- Median Daily Consumption: ~7.05 units
- Minimum Consumption: 5.2 units
- Maximum Consumption: 10.5 units
- High Usage Days (>9 units): Identified
- Estimated Monthly Bill: ₹1348.2

#### Simple Insights

- Most days have normal electricity usage
- Only a few days use very high electricity
- High-usage days increase the total bill
- Some days show possible electricity wastage
- Electricity usage pattern is mostly consistent

#### Business Impact

- Helps reduce electricity costs by identifying high usage and wastage days
- Supports better budgeting and monthly bill estimation
- Improves energy efficiency and operational planning
- Enables data-driven decisions for cost optimization
- Promotes sustainable and efficient electricity usage